src/pages/Detail/component/ReviewModal.jsx

```diff
@@ -0,0 +1,248 @@
+  const [reviewTxt, setReviewTxt] = useState('');
+  const nextId = useRef(4);
+  const token = localStorage.getItem('token');
+
+  const previewHandle = e => {
+    e.preventDefault();
+
+    if (e.target.files.length) {
+      const imgTarget = e.target.files[0];
+      const fileReader = new FileReader();
+
+      fileReader.readAsDataURL(imgTarget);
+      fileReader.onload = e => {
+        setImgSrc(e.target.result);
+        setimgFile(e.target.result);
+      };
+    } else {
+      setImgSrc('');
+    }
+  };
+
+  const removeHandle = e => {
+    e.preventDefault();
+    setImgSrc('');
+  };
+
+  const reviewOnChange = e => {
+    setReviewTxt(e.target.value);
+  };
+
+  const OnSubmit = e => {
+    e.preventDefault();
+
+    if (!token) {
+      const reviewLength = reviewTxt.length;
+
+      if (reviewLength >= 200) {
+        alert('200자 이하로 입력해주세요');
+      } else if (reviewLength < 10) {
+        alert('10자 이상으로 입력해주세요');
+      } else {
+        const review = {
+          review_id: nextId.current,
+          content: reviewTxt,
+          image_url: imgSrc,
+        };
+
+        setReviews(reviews.concat(review));
+        nextId.current += 1;
+
+        closeModal();
+      }
+    } else {
+      alert('로그인 후 이용해주세요');
+
+      const formData = new FormData();
+
+      formData.append('reviewImage', imgFile);
+      formData.append('reviewText', reviewTxt);
+
+      fetch('API 주소', {
+        method: 'POST',
+        headers: {
+          Authorization: token,
+          enctype: 'multiple/form-data',
+        },
+        body: formData,
+      })
+        .then(response => response.json())
+        .then(result => {
+          if (result.status === '200') {
+            closeModal();
+          } else {
+            alert('리뷰 등록에 실패했습니다');
+          }
+        });
+    }
+  };
+
+  return (
+    <ModalChang>
+      <LetsReview>글 작성하기</LetsReview>
+
+      <form>
+        <Wrap>
+          <UserClass>
+            <UserImg />
+            <ClassName>{classTitle}</ClassName>
+          </UserClass>
+          <PhotoLabel htmlFor="inputFile">
+            <HiOutlinePhotograph
+              size="20"
+              style={{ position: 'relative', top: '5px', marginRight: '3px' }}
+            />
+            사진 첨부
+          </PhotoLabel>
+          <input
+            id="inputFile"
+            type="file"
+            name="reviewImage"
+            accept="image/*"
+            style={{ display: 'none' }}
+            onChange={previewHandle}
+          />
+        </Wrap>
+
+        <WriteZone>
+          {imgSrc && (
+            <ImgPreview>
+              <ImgRemoveBtn onClick={removeHandle}>x</ImgRemoveBtn>
+              <img src={imgSrc} alt="preview-img" />
+            </ImgPreview>
+          )}
+          <Textarea
+            type="text"
+            name="reviewText"
+            maxLength={201}
+            placeholder="무단 홍보, 비방 글은 삭제 될 수 있습니다."
+            onChange={reviewOnChange}
```

Comment:
```suggestion
            onChange={(e) => setReviewTxt(e.target.value)}
```
state를 변경하는 함수 하나만 사용된다면 위와 같이 작성하는 것이 더 효율적입니다.